### ISSUE TITLE
SSL Version Error on Installation

### ISSUE THREAD

**Issue by mikesbots** [NSGeophysics/GPRPy#24]
Hi All

Whilst working through the installation, after successfully downloading and extracting the software I successfully run the "python installMigration.py" command but when doing the "pip install ." I get a SSLError as follows...

___________________________ 
C:\Users\mikes\Desktop\GPRPy-master>pip install .
Processing c:\users\mikes\desktop\gprpy-master
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)'))': /simple/tqdm/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)'))': /simple/tqdm/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)'))': /simple/tqdm/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)'))': /simple/tqdm/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)'))': /simple/tqdm/
Could not fetch URL https://pypi.org/simple/tqdm/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/tqdm/ (Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)'))) - skipping
ERROR: Could not find a version that satisfies the requirement tqdm (from gprpy) (from versions: none)
ERROR: No matching distribution found for tqdm
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)'))) - skipping
________________________________________________________________________________________________________ 
I upgraded to python 3.10.1 and am running python from an Anaconda3 command prompt.  I've noticed that a lot of other users have similar errors when trying to do any pip install, but none of the suggested fixed (disable firewall and so on) have worked for me.  Has anyone else had this issue and is there a fix or a workaround.  I tried installing Git and running from there but it gave me exactly the same error.

Any help appreciated.

Mike

**Comment by AlainPlattner** [comment]
Hi Mike,

This looks a bit as if tqdm may not yet be available for python 3.10.1.

An "easy" solution could be to install python 3.9 and install GPRPy for
that version. Anaconda males having several versions of python quite easy.

Best,
Alain

**Comment by mikesbots** [comment]
Thanks Alain

In the end it turned out to be some or other quirky connection setting specific to one of my laptops and not tqdm non-availability in 3.10.1.  The solution I found by accident was to switch my proxy on and then off again and then I have about 10 seconds to type the "pip install ." on the command prompt and there is a brief window where it doesn't throw an SSL error.  It would install a couple of things and then trip up and throw me the same error but if I switched the proxy on then off and repeated it would pick up where it left off and install a few more components.  Eventually after about 5 attempts it worked through the complete install and I had everything I needed to get going and I now have a working version of GPRPy.

Not entirely satisfactory in the sense that I still don't really know what the issue is but at least I have a clunky work around in the interim and it's me that's the problem, not the entire world.

Regards

Mike
